### PR TITLE
docs: Add structure to master config ref

### PR DESCRIPTION
**Pull request by tara-det-ai**
Add structure to master config ref, add heading levels, to enable better search results

### CODE REVIEW

File: docs/reference/reference-deploy/config/master-config-reference.rst [hunk 1]
```diff
@@ -4,957 +4,1565 @@
...
-      Please only specify either ``connection_string`` or the ``account_url`` and ``credential``
-      pair.
+``spot_max_price``
+^^^^^^^^^^^^^^^^^^
 
-   -  ``type: shared_fs``: Checkpoints are written to a directory on the agent's file system. The
-      assumption is that the system administrator has arranged for the same directory to be mounted
-      at every agent host, and for the content of this directory to be the same on all agent hosts
-      (e.g., by using a distributed or network file system such as GlusterFS or NFS).
+Optional field indicating the maximum price per hour that you are willing to pay for a spot
```
Comment (eecsliu) [inline docs/reference/reference-deploy/config/master-config-reference.rst:963]: Nit: seems like `Required.` is used a lot, but `Optional.` isn't used as much. Not sure if there is guidance on styling for this, but I'll leave that up to you.

**Review by eecsliu — APPROVED** [review]
Looks good! The only thing I would change is the first wording comment, but the rest is in good state

File: docs/reference/reference-deploy/config/master-config-reference.rst [hunk 2]
```diff
@@ -4,957 +4,1565 @@
...
-The Determined master supports a range of configuration settings that can be set via a :ref:`YAML
+The Determined master supports various configuration settings that can be set via a :ref:`YAML
 configuration file <topic-guides_yaml>`, environment variables, or command-line options. The
-configuration file is normally located at ``/etc/determined/master.yaml`` on the master and is read
-when the master starts.
+configuration file is typically located at ``/etc/determined/master.yaml`` on the master and it is
+read when the master starts up.
 
-The configuration of an active master can be examined using the Determined CLI with the command
+To inspect the configuration of an active master, use the Determined CLI by executing the command
```
[the remark this answers is not in the document]
Comment (tara-det-ai) [reply docs/reference/reference-deploy/config/master-config-reference.rst:12]: good catch!

File: docs/reference/reference-deploy/config/master-config-reference.rst (hunk 1 above)
Comment (tara-det-ai) [reply docs/reference/reference-deploy/config/master-config-reference.rst:963]: thank you!

**Pull request merged**